includes/payments/functions.php

```diff
@@ -1063,9 +1077,36 @@ function edd_get_payment_user_email( $payment_id ) {
  * @return string $user_id User ID
  */
 function edd_get_payment_user_id( $payment_id ) {
-	$user_id = edd_get_payment_meta( $payment_id, '_edd_payment_user_id', true );
+	$user_id = 0;
+
+	// check the customer record first
+	$customer_id = edd_get_payment_customer_id( $payment_id );
+	$customer    = new EDD_Customer( $customer_id );
+
+	if ( ! empty( $customer->user_id ) && $customer->user_id > 0 ) {
+		$user_id = $customer->user_id;
+	}
+
+	// check the payment meta if we're still not finding a user with the customer record
+	if ( ! empty( $user_id ) && $user_id > 0 ) {
```

Comment:
No need for `$user_id > 0` check here since it's only getting set to something other than zero if the ID found on line 1083 s greater than 0.